[Rachel Tan] iP

superidol2 opened the pull request:
# Awebo Chatbot Overview 🐧

Awebo is a smart and intuitive task management chatbot that helps users stay organized through a simple command-line interface. Inspired by the call of a ptarmigan, Awebo is always ready to keep your tasks in check!

## Key Features

- **Command parsing and delegation** for seamless task management
- **Multiple task types**:
  - To-Do tasks
  - Deadlines with date and time
  - Events with a start and end time
- **Persistent storage** to ensure your tasks are never lost

## Components Overview

- **DateFormatter**: Manages date and time formatting for deadlines and events.
- **Parser**: Interprets user commands and directs them to the appropriate functions.
- **Storage**: Saves and loads tasks from `./data/tasks.txt` to maintain continuity.
- **UI**: Handles user input and output, keeping interactions natural and efficient.
- **Task Tracker and Marker**: Tracks tasks and marks them as completed.
- **List**: Displays all tasks in an organized manner.

---

## Sample Code

```java
package awebo;
import java.io.IOException;

import awebo.gui.MainWindow;
import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.scene.layout.AnchorPane;
import javafx.stage.Stage;

/**
 * A GUI for Duke using FXML.
 */
public class Main extends Application {

    private Awebo awebo = new Awebo();

    @Override
    public void start(Stage stage) {
        try {
            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
            AnchorPane ap = fxmlLoader.load();
            Scene scene = new Scene(ap);
            stage.setScene(scene);
            // inject the awebo instance
            fxmlLoader.<MainWindow>getController().setAwebo(awebo);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

superidol2 committed:
Add Awebo.java

superidol2 committed:
Add echo to Awebo.java

superidol2 committed:
Add add and list to Awebo.java

superidol2 committed:
Add mark as done to Awebo.java

superidol2 committed:
Add todo, deadline, event to Awebo.java

superidol2 committed:
Change the project name in README

superidol2 committed:
Change the name from Duke to Awebo in runtest.sh

superidol2 committed:
Change the expected text in EXPECTED.TXT

superidol2 committed:
Remove the extra space in list command

superidol2 committed:
Modify the test cases for multiple commands

superidol2 committed:
add error handling for empty tasks in Awebo.java

superidol2 committed:
add delete task to Awebo.java

superidol2 committed:
Add save function in branch

superidol2 committed:
Merge branch 'branch-Level-7'

superidol2 committed:
Add datetime to tasks in Awebo.java

superidol2 committed:
Merge branch 'branch-Level-8'

Pranavan135 commented on src/main/java/Awebo.java:
```diff
@@ -0,0 +1,170 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+abstract class Task {
```
There should be at least one line after the import statement

Pranavan135 replied:
Class should have a header comment

Pranavan135 commented on src/main/java/Awebo.java:
```diff
@@ -0,0 +1,170 @@
+    }
+}
+
+class ToDo extends Task {
```
Class is missing a header comment. Refer our coding standard.

Pranavan135 commented on src/main/java/Awebo.java:
```diff
@@ -0,0 +1,170 @@
+        System.out.println("Hello! I'm Awebo\nWhat can I do for you?\n");
+        while (true) {
+            String userinput = scanner.nextLine();
+            if (userinput.equalsIgnoreCase("bye")) { //case-insensitive check
```
Use constants for Strings such as "bye". It makes the code more readable.

Pranavan135 commented on src/main/java/Awebo.java:
```diff
@@ -0,0 +1,170 @@
+        this.isDone = true;
+    }
+    public void markUndone() {
+        this.isDone = false;
```
Use the correct function name for getter and setter

Pranavan135 commented on src/main/java/Awebo.java:
```diff
@@ -0,0 +1,170 @@
+        this.isDone = false;
+    }
+    public String getStatus() {
+        return (isDone ? "[X] " : "[ ] ") + description;
```
Avoid one line if statement. It makes the code less readable.

Pranavan135 reviewed (commented):
The implementation should follow the coding standard. The implementation is missing comments. In addition, the naming should follow the coding standard. 

In addition, it is better to keep a single java class in a single file. i.e. It is not advisable to have too many classes in a single file.

superidol2 committed:
Refactor classes into OOP

superidol2 committed:
Merge branch 'branch-A-MoreOOP'

superidol2 committed:
Refactor files to package

superidol2 committed:
Merge branch 'add-gradle-support' into branch-add-gradle-support

superidol2 committed:
Delete saved_tasks folder

superidol2 committed:
Modify .gitignore

superidol2 committed:
Modify class name in gradle

superidol2 committed:
Merge branch 'branch-add-gradle-support'

superidol2 committed:
Add JUnit Tests

superidol2 committed:
Add JavaDoc

superidol2 committed:
Add checkstyle

superidol2 committed:
Change format of switch case

superidol2 committed:
Change public variable to private with get function

superidol2 committed:
Change public variables to private with get functions

superidol2 committed:
Change order of imports

superidol2 committed:
Change public variables to private with get functions

superidol2 committed:
Add new line after code

superidol2 committed:
Change order of imports

superidol2 committed:
Add find task function

superidol2 committed:
Add test.txt to .gitignore

superidol2 committed:
Add cheer to Parser

superidol2 committed:
Merge branch 'branch-A-JavaDoc'

superidol2 committed:
Merge branch-A-CodingStandard to master

superidol2 committed:
Remove duplicate variable in Event.java

superidol2 committed:
Merge branch 'branch-Level-9'

superidol2 committed:
Merge branch-A-Cheer to master

superidol2 committed:
Add checkstyle branch

superidol2 committed:
Merge pull request #1 from superidol2/branch-A-CheckStyle

Add checkstyle branch

superidol2 committed:
Add JavaFX to build.gradle

superidol2 committed:
Add GUI elements to app

superidol2 committed:
Modify Parser and Ui for list command

superidol2 committed:
Update gradle

superidol2 committed:
Merge pull request #2 from superidol2/branch-Level-10

Branch level 10

superidol2 committed:
Improved date formatter

superidol2 committed:
Improved event error handling

superidol2 committed:
Add welcome message and help command

superidol2 committed:
Rewrite parser unit tests

superidol2 committed:
Merge pull request #3 from superidol2/branch-A-UnitTesting

Branch A unit testing

PerRucia commented on src/main/java/awebo/parser/Parser.java:
```diff
@@ -0,0 +1,251 @@
+            Storage.writeListToFile(list, filePath);
+        } catch (Exception e) {
+            ui.showMessage("Invalid task number.");
+        }
```
Maybe multiple catch statements would be more suitable for debugging as the try block can throw multiple exception types.

PerRucia commented on src/main/java/awebo/parser/Parser.java:
```diff
@@ -0,0 +1,251 @@
+    private void markTask(String command, Ui ui) {
+        try {
+            int index = Integer.parseInt(command.substring(5)) - 1;
+            list.get(index).markDone();
```
Maybe predefined constants should be used here instead of magic numbers?

PerRucia commented on src/main/java/awebo/task/Task.java:
Javadoc comments are consistently present for every public method across the repository, which greatly improves code readability and makes reviewing and understanding the codebase more efficient.

superidol2 committed:
Add assert feature

superidol2 committed:
Added assertions to validate inputs and enforce expected conditions, improving code reliability and debugging.

- Added assertions in parser function to validate input parameters.
- Improved error messaging for assertion failures.

superidol2 committed:
Implement duplicate task checker in the Parser class
- Added a method `isDuplicate(String taskDescription)` to check for duplicate tasks before adding a new task to the list.
- This implementation ensures that users cannot add duplicate tasks until they are removed, improving the overall robustness of task management in the awebo app.

superidol2 committed:
Merge pull request #5 from superidol2/branch-C-DetectDuplicates

Implement duplicate task checker in the Parser class

superidol2 committed:
Modify command handling logic

superidol2 committed:
Update unit tests for task handling

superidol2 committed:
Refine UI for a cleaner and more polished appearance

superidol2 committed:
Merge pull request #7 from superidol2/branch-A-BetterGui

Refine UI for a cleaner and more polished appearance

superidol2 committed:
Add Javadoc comments for better code documentation

superidol2 committed:
Add Ui.png image to docs for UI reference

superidol2 committed:
Add Ui.png image to Awebo User Guide section

superidol2 committed:
Update README with Awebo User Guide and Table of Contents

superidol2 committed:
Update README with revised content and formatting improvements

superidol2 committed:
Fix import wildcard in DialogBox

superidol2 committed:
Add load task to storage

superidol2 committed:
Merge pull request #8 from superidol2/branch-fix-storage

Branch fix storage

superidol2 committed:
Update parser parameter in Awebo.java

superidol2 committed:
Update user guide
